Automattic/jetpack
# de-Fusioning: Prepare for re-syncing class.jetpack-user-agent.php

#### Changes proposed in this Pull Request:

These haven't been synced since 2020, when a rearrangement of the code put the `jetpack_is_mobile()` function into a place where it couldn't be loaded early enough for certain callers on wpcom.

Now that we're about to de-Fusion it, we also need to take into account that the autoloader won't be set up to load the classes from the device-detection package that early either.

And finally, since there are uses of the deprecated Jetpack_User_Agent_info class spread across 5 different repos on wpcom, let's not raise deprecation warnings when running there just yet.

#### Other information:


#### Jetpack product discussion


https://github.com/Automattic/jetpack/issues/25213#issuecomment-1285897313
p1666286361466369-slack-C032DVAHMK3

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Make sure whatever uses this stuff still works.
* Load this and D90369-code and D90372-code onto a wpcom sandbox. Make sure stuff still works there too.
